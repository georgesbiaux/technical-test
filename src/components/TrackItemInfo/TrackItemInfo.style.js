import styled from 'styled-components';

const StyledTrackItemInfo = styled.div`
  display: flex;
  flex-direction: row;

  .track-name-and-artist {
    text-align: left;
    display: flex;
    flex-direction: column;
    padding-left: 5px;
  }

  .track-name {
    font-size: 14px;
    line-height: 18px;
    color: #343434;
  }

  .track-artist {
    font-size: 12px;
    line-height: 18px;
    color: #8f8f8f;
  }
`;

export default StyledTrackItemInfo;
