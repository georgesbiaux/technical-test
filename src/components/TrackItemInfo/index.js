export { default } from './TrackItemInfo.component';
