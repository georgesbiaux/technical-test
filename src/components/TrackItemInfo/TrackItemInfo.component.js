import React from 'react';
import { TrackItemCover } from 'components';
import StyledTrackItemInfo from './TrackItemInfo.style';

const TrackItemInfo = props => (
  <StyledTrackItemInfo className="track-info">
    <TrackItemCover pictureUrl={props.track.pictureUrl} />
    <div className="track-name-and-artist">
      <div className="track-name">
        {props.track.name}
      </div>
      <div className="track-artist">
        {props.track.artist}
      </div>
    </div>
  </StyledTrackItemInfo>
);

export default TrackItemInfo;
