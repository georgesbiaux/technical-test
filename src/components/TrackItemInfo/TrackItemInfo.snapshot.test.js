import React from 'react';
import { render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TrackItemInfo from './TrackItemInfo.component';

describe('<TrackItemInfo>', () => {
  it('should render correctly', () => {
    const props = {
      track: {
        id: 1,
        name: 'Beast Of Burden',
        duration: 211,
        priority: false,
        artist: 'The Rolling Stones',
        adder: {
          id: 1,
          name: 'kant',
          pictureUrl:
            'https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd',
        },
        pictureUrl:
          'https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372',
        votes: {
          count: 0,
          userVoted: false,
        },
      },
    };
    const tree = render(<TrackItemInfo {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
