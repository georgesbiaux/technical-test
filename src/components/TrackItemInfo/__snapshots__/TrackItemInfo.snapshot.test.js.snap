// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackItemInfo> should render correctly 1`] = `
<div
  class="track-info sc-iwsKbI ecisHD"
>
  <div
    class="sc-EHOje bMqQgi"
  >
    
  </div>
  <div
    class="track-name-and-artist"
  >
    <div
      class="track-name"
    >
      Beast Of Burden
    </div>
    <div
      class="track-artist"
    >
      The Rolling Stones
    </div>
  </div>
</div>
`;
